src/setup.py

```diff
@@ -8,7 +8,14 @@
         "pdist_agregation",
         sources=["_pdist_agregation.pyx"],
         include_dirs=[numpy.get_include()],
-        extra_compile_args=['-fopenmp'],
+        extra_compile_args=[
+            '-fopenmp',
+            '-O3',
+            '-ftree-vectorize',
```

Comment:
It would be interesting just to try '-03' and '-ftree-vectorize' to see if we get the same performance.

Reply:
They are being reran with fdc9d31 on a smaller config.